message_list: Improve "not subscribed" banner.

Aadithya-J opened the pull request:
Update the trailing bookend to include "Load updates" and "Subscribe" buttons.

Subscribing to a stream being viewed now fetches newer messages.

There was earlier work on this issue in #36133. I chose to start from scratch rather than building on that PR because :

* The "Load updates" buttion functionality did not correctly fetch newer messages.
* The UI styling did not match the issue description

Fixes: https://github.com/zulip/zulip/issues/35901

**How changes were tested:**



<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

https://github.com/user-attachments/assets/122043f4-8b62-40f9-ad1b-ec1b35a56fd7

<table border="1">
  <thead>
    <tr>
      <th>Image Name</th>
      <th>Screenshot</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Lock</td>
      <td><img width="1437" height="731" alt="Screenshot 2025-12-04 at 12 35 43 PM" src="https://github.com/user-attachments/assets/e1d39936-47c8-4135-9b2f-f10acb41e175" /></td>
    </tr>
    <tr>
      <td>Hashtag</td>
      <td><img width="1440" height="736" alt="Screenshot 2025-12-04 at 12 35 31 PM" src="https://github.com/user-attachments/assets/58e0612a-2d8a-4362-a8bf-f5b349b80f3e" /></td>
    </tr>
    <tr>
      <td>Globe</td>
      <td><img width="1439" height="732" alt="Screenshot 2025-12-04 at 12 35 23 PM" src="https://github.com/user-attachments/assets/63d3e5e8-8671-4255-a075-a391ae3b7f30" /></td>
    </tr>
  </tbody>
</table>

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
Thanks! Please follow the guidelines for how to [present your pull request](https://zulip.readthedocs.io/en/latest/contributing/continuing-unfinished-work.html#present-your-pull-request) when there is prior work on an issue.

alya commented:
I also noticed that there's not enough space between "to" and the channel privacy icon.

alya commented:
Thanks! Can you please add screenshots demonstrating the other types of privacy icons (lock and globe)?

Aadithya-J commented:
updated to include the screenshot with the globe icon.
my understanding is that the lock icon should not appear here because private channels aren’t accessible to unsubscribed users.please correct me if I’m mistaken.

Aadithya-J commented:
I've added the screenshot with the lock icon included. As far as I can tell, there isn't another banner that uses privacy icons for reference. I’ve updated the icons to match the font size of the left sidebar and adjusted the alignment. Please have a look

alya commented:
OK, thanks! The screenshots look OK to me now.

@amanagr Will review this PR after merge conflicts are resolved.

amanagr commented on web/src/click_handlers.ts:
```diff
@@ -298,6 +278,17 @@ export function initialize(): void {
         window.location.href = this.href;
     });
 
+    $("#main_div").on("click", ".load-newer-messages-button", (e) => {
```
For selector you can use `$("body").on("click", ".not-subscribed-banner .load-newer-messages-button"` to better indicate the context of the button.

amanagr commented on web/src/click_handlers.ts:
```diff
@@ -298,6 +278,17 @@ export function initialize(): void {
+            message_fetch.maybe_load_newer_messages({
+                msg_list: message_lists.current,
+                force: true,
+            });
```
This will not work to show the correct narrow state. You need to call message_view.show(CURRENT_NARROW) with the correct `then_select_id` and `then_select_offset` and `force_rerender`.
